### PR TITLE
Adding the fail_limit option for push and checkout

### PR DESCRIPTION
**Pull request by lucasncabral**
This option aims to provide the user with a way to define a limit of failures that the process may have before being aborted [default: no limit - ML-Git will be trying to send all the blobs until the list is finished].

```
Usage: ml-git datasets push [OPTIONS] ML_ENTITY_NAME

  Push local commits from ML_ENTITY_NAME to remote ml-git repository &
  storage.

Options:
  --retry INTEGER       Number of retries to download the files from the
                        storage [default: 2].
  --clearonfail         Remove the files from the storage in case of failure
                        during the push operation.
 --fail-limit INTEGER   Number of failures before aborting the command 
                        [default: no limit].
  --verbose             Debug mode
```

In addition, modifications were made to summarize the errors (the current behavior of ML-Git is to present a massive amount of messages).

Example:
```
$ ml-git datasets push dataset-ex --fail-limit=20
files: 17%|███████                                                                                   | 1.66k/10.0k [01:05<03:59, 35 files/s, Failed=20]
ERROR - MLGit: There was an error sending the data. 8375 pending blobs to send.
ERROR - MLGit: ERROR FOUND: EndpointConnectionError - Could not connect to the endpoint URL: "http://127.0.0.1:9000/mlgit-test-lucas/zdj7WisdDH58bEju9co43GqjZfnoi8amu"
ERROR - MLGit: It was not possible to recover from the error found. Please fix the problem and rerun the command again.
```

### CODE REVIEW

File: ml_git/error_handler.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    handler = {}
+    for ep in pkg_resources.iter_entry_points(group=entry_point):
+        handler.update({ep.name: ep.load()})
+    exit_code = 1
+    if len(handler) > 0 and error_handler_method_name in handler:
+        exit_code = handler[error_handler_method_name](error)
+    return exit_code
+
+
+def error_handler(errors_count, error, push_method=True):
```
Comment (gilMars) [inline ml_git/error_handler.py:27]: I think that this message can be extracted to the method that call the error_handler.
`def error_handler(errors_count, error, log_error_message):`

File: docs/mlgit_commands.md [hunk 2]
```diff
@@ -104,24 +104,27 @@ Both are the same representation. One is human-readable and is also used interna
 <br>
 
 ```
-Usage: ml-git models checkout [OPTIONS] ML_ENTITY_TAG|ML_ENTITY
+Usage: ml-git models checkout [OPTIONS] ML_ENTITY_TAG
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply docs/mlgit_commands.md]: It was unintentionally removed. Added again on 2e76e60.

File: ml_git/error_handler.py (hunk 1 above)
Comment (lucasncabral) [reply ml_git/error_handler.py]: I agree. The logs have been removed from the method at 2e76e60.

File: docs/api/README.md [hunk 3]
```diff
@@ -52,9 +52,11 @@ def checkout(entity, tag, sampling=None, retries=2, force=False, dataset=False,
         force (bool, optional): Force checkout command to delete untracked/uncommitted files from the local repository [default: False].
         dataset (bool, optional): If exist a dataset related with the model or labels, this one must be downloaded [default: False].
         labels (bool, optional): If exist labels related with the model, they must be downloaded [default: False].
-    
+        fail_limit (int, optional): Define a limit of failures that the process can have before it is aborted [default: no limit].
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply docs/api/README.md]: Suggestion applied in 264d505.

**Review by gilMars — APPROVED** [review]

**Review by andersonpablosilva — APPROVED** [review]

File: docs/api/README.md [hunk 4]
```diff
@@ -52,9 +52,11 @@ def checkout(entity, tag, sampling=None, retries=2, force=False, dataset=False,
         force (bool, optional): Force checkout command to delete untracked/uncommitted files from the local repository [default: False].
         dataset (bool, optional): If exist a dataset related with the model or labels, this one must be downloaded [default: False].
         labels (bool, optional): If exist labels related with the model, they must be downloaded [default: False].
-    
+        fail_limit (int, optional): Failure limit to abort the command [default: no limit].
```
Comment (gpretto) [inline docs/api/README.md:55]:
```suggestion
        fail_limit (int, optional): Number of failures before aborting the command [default: no limit].
```

File: docs/api/README.md [hunk 5]
```diff
@@ -186,7 +188,8 @@ def push(entity, entity_name,  retries=2, clear_on_fail=False):
             entity_name (str): An ml-git entity name to identify a ML entity.
             retries (int, optional): Number of retries to upload the files to the storage [default: 2].
             clear_on_fail (bool, optional): Remove the files from the storage in case of failure during the push operation [default: False].
-         """
+            fail_limit (int, optional): Failure limit to abort the command [default: no limit].
```
Comment (gpretto) [inline docs/api/README.md:191]:
```suggestion
            fail_limit (int, optional): Number of failures before aborting the command [default: no limit].
```

File: docs/mlgit_commands.md [hunk 6]
```diff
@@ -109,19 +109,22 @@ Usage: ml-git models checkout [OPTIONS] ML_ENTITY_TAG|ML_ENTITY
...
+  --retry INTEGER       Number of retries to download the files from the
+                        storage [default: 2].
+  --force               Force checkout command to delete untracked/uncommitted
+                        files from local repository.
+  --bare                Ability to add/commit/push without having the ml-
+                        entity checked out.
+  --version INTEGER     Number of artifact version to be downloaded [default:
+                        latest].
+  --fail-limit INTEGER  Failure limit to abort the command [default: no
+                        limit].
```
Comment (gpretto) [inline docs/mlgit_commands.md:125]:
```suggestion
  --fail-limit INTEGER  Number of failures before aborting the command [default: no limit].
```

File: docs/mlgit_commands.md [hunk 7]
```diff
@@ -474,11 +477,13 @@ Usage: ml-git datasets push [OPTIONS] ML_ENTITY_NAME
...
-                   storage [default: 2].
-  --clearonfail    Remove the files from the storage in case of failure during
-                   the push operation.
-  --verbose        Debug mode
+  --retry INTEGER       Number of retries to download the files from the
+                        storage [default: 2].
+  --clearonfail         Remove the files from the storage in case of failure
+                        during the push operation.
+  --fail-limit INTEGER  Failure limit to abort the command [default: no
+                        limit].
```
Comment (gpretto) [inline docs/mlgit_commands.md:485]:
```suggestion
  --fail-limit INTEGER  Number of failures before aborting the command [default: no limit].
```

File: ml_git/api.py [hunk 8]
```diff
@@ -61,6 +61,7 @@ def checkout(entity, tag, sampling=None, retries=2, force=False, dataset=False,
         force (bool, optional): Force checkout command to delete untracked/uncommitted files from the local repository [default: False].
         dataset (bool, optional): If exist a dataset related with the model or labels, this one must be downloaded [default: False].
         labels (bool, optional): If exist labels related with the model, they must be downloaded [default: False].
+        fail_limit (int, optional): Failure limit to abort the command [default: no limit].
```
Comment (gpretto) [inline ml_git/api.py:64]:
```suggestion
        fail_limit (int, optional): Number of failures before aborting the command [default: no limit].
```

File: ml_git/api.py [hunk 9]
```diff
@@ -182,10 +184,11 @@ def push(entity, entity_name,  retries=2, clear_on_fail=False):
             entity_name (str): An ml-git entity name to identify a ML entity.
             retries (int, optional): Number of retries to upload the files to the storage [default: 2].
             clear_on_fail (bool, optional): Remove the files from the storage in case of failure during the push operation [default: False].
+            fail_limit (int, optional): Failure limit to abort the command [default: no limit].
```
Comment (gpretto) [inline ml_git/api.py:187]:
```suggestion
            fail_limit (int, optional): Number of failures before aborting the command [default: no limit].
```

File: ml_git/ml_git_message.py [hunk 10]
```diff
@@ -288,6 +289,10 @@
     'ERROR_INVALID_SPEC_VALUE_IN': 'Invalid %s spec in %s.  It should look something like this:\n%s',
     'ERROR_SPEC_FILE_NOT_FOUND': '\nCan\'t find  spec file to increment version.  Are you in the '
                                   'root of the repo?\n     %s\n',
+    'ERROR_ON_PUSH_BLOBS': 'There was an error sending the data. %s pending blobs to send.',
+    'ERROR_ON_GETTING_BLOBS': 'There was an error downloading the data. %s pending blobs to get.',
+    'ERROR_CANNOT_RECOVER': 'It was not possible to recover from the error found. Please fix the problem and run the command again.',
```
Comment (gpretto) [inline ml_git/ml_git_message.py:294]:
```suggestion
    'ERROR_CANNOT_RECOVER': 'It was not possible to recover from the error found. Please fix the problem and rerun the command again.',
```

**Review by gpretto — COMMENTED** [review]
Minor changes.

File: ml_git/ml_git_message.py (hunk 10 above)
Comment (lucasncabral) [reply ml_git/ml_git_message.py]: Suggestion applied in 85a9fa3.

**Review by gpretto — APPROVED** [review]
Good!

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**